gtsfm/utils/io.py

```diff
@@ -85,3 +84,79 @@ def save_json_file(
         json.dump(data, f, indent=4)
 
 
+def read_bal(file_path: str) -> GtsfmData:
+    """Read a BAL file.
+
+    Args:
+        file_name: file path of the BAL file.
+
+    Returns:
+        The data as an GtsfmData object.
+    """
+    sfm_data = gtsam.readBal(file_path)
+
+    num_images = sfm_data.number_cameras()
+
+    gtsfm_data = GtsfmData(num_images)
+    for i in range(num_images):
+        camera = sfm_data.camera(i)
+        gtsfm_data.add_camera(camera, i)
+    for j in range(sfm_data.number_tracks()):
+        gtsfm_data.add_track(sfm_data.track(j))
+
+    return gtsfm_data
+
+
+def write_cameras(gtsfm_data: GtsfmData, save_dir: str) -> None:
+    """Writes the camera data file in the COLMAP format.
+
+    Args:
+        gtsfm_data: scene data to write.
+        save_dir: folder to put the cameras.txt file in.
+    """
+    os.makedirs(save_dir, exist_ok=True)
+
+    # TODO: get image shape somehow
+    image_width = 1000  # pylint: disable=unused-variable
+    image_height = 1000  # pylint: disable=unused-variable
+
+    # TODO: handle shared intrinsics
+
+    file_path = os.path.join(save_dir, "cameras.txt")
+    with open(file_path, "w") as f:
+        f.write("# Number of cameras: {}\n".format(gtsfm_data.number_images()))
+
+        for i in gtsfm_data.get_valid_camera_indices():
+            camera = gtsfm_data.get_camera(i)
+            calibration = camera.calibration()
+
+            fx = calibration.fx()
+            u0 = calibration.px()
+            v0 = calibration.py()
+            k1 = calibration.k1()
+            k2 = calibration.k2()
+
+            f.write("{} SIMPLE_RADIAL {} {} {} {} {} {} {}\n".format(i, image_width, image_height, fx, u0, v0, k1, k2))
```

Comment:
i think python f-strings would make this even simpler :-)

Reply:
Oh I forgot to put `f` earlier and that is why it was not working. Will revert and fix that.